plugins/cpp/service/src/filediagram.cpp

```diff
@@ -458,28 +477,50 @@ std::vector<util::Graph::Node> FileDiagram::getDependFiles(
   const util::Graph::Node& node_,
   bool reverse_)
 {
-  std::vector<util::Graph::Node> depends;
+  std::vector<core::FileId> depends;
 
   _transaction([&, this]{
-    EdgeResult res = _db->query<model::CppEdge>(
-      (reverse_
-         ? EdgeQuery::to->domainId
-         : EdgeQuery::from->domainId) == node_ &&
-      EdgeQuery::from->domainId != EdgeQuery::to->domainId &&
-      EdgeQuery::type == model::CppEdge::DEPEND);
+    auto contained = _db->query<model::File>(
+      odb::query<model::File>::parent == std::stoull(node_) &&
+      odb::query<model::File>::type != model::File::DIRECTORY_TYPE
+      );
 
-    for (const model::CppEdge& edge : res)
+    std::unordered_set<core::FileId> used;
+
+    for(const model::File& file : contained)
```

Comment:
Generally the same comments apply here as for the other method.

Reply:
See above